### DISCUSSION TITLE
Questions Regarding the 'SafetyPointMultiGoal1-v0' Environment

### DISCUSSION THREAD

**Discussion by JingjieNi** [PKU-Alignment/safety-gymnasium#129] [Q&A]
I am interested in your multi-agent safety reinforcement learning environments. I have carefully reviewed the code related to 'SafetyPointMultiGoal1-v0' and currently have two points of confusion.

1. In this environment, both the observation and state include the radar information for two agents. Considering the overall consistency of the environment, the radar information for different agents seems to only change the angle of observation of the environment?

2. From the perspective of cost['agent_0']['cost_contact_other'], it appears that agent 1 needs to avoid obstacles with agent 0. However, it seems that agent 1's observation does not include a radar facing agent 0?

I am not sure if my understanding is correct and would appreciate the opportunity to have a further discussion with the code authors.
![图片1](https://github.com/user-attachments/assets/2d5fce49-60fe-44ba-96f8-982b51055d8a)

**Comment by muchvo** [comment]
1. Yes, essentially, it centers on different agents observing in various directions, so the core difference is in the angles.

2. Yes, for vector inputs, radar observations of the other agent should be included, whereas this isn’t an issue for vision inputs. Before implementing this update, we should consider disabling the `contact_other_cost` option when using vector inputs.

Thank you for your insight!
